src/main/java/net/dv8tion/jda/core/handle/ChannelUpdateHandler.java

```diff
@@ -115,15 +117,28 @@ protected Long handleInternally(JSONObject content)
                 //Get the current overrides. (we copy them to a new list because the Set returned is backed by the Map, meaning our removes would remove from the Map. Not good.
                 //Loop through all of the json defined overrides. If we find a match, remove the User or Role from our lists.
                 //Any entries remaining in these lists after this for loop is over will be removed from the Channel's overrides.
-                List<Role> collect = channel.getRoleOverrideMap().keySet().stream().filter(role -> !containedRoles.contains(role)).collect(Collectors.toList());
-                collect.forEach(role -> {
-                    changedRoles.add(role);
-                    channel.getRoleOverrideMap().remove(role);
-                });
-                List<Member> collect1 = channel.getMemberOverrideMap().keySet().stream().filter(user -> !containedMembers.contains(user)).collect(Collectors.toList());
-                collect1.forEach(member -> {
-                    changedMembers.add(member);
-                    channel.getMemberOverrideMap().remove(member);
+                TLongList toRemove = new TLongLinkedList();
+                TDecorators.wrap(channel.getOverrideMap().keySet()).stream()
+                        .map(id -> channel.getGuild().getRoleById(id))
+                        .filter(Objects::nonNull)
+                        .filter(role -> !containedRoles.contains(role))
+                        .forEach(role -> {
+                            changedRoles.add(role);
+                            toRemove.add(role.getIdLong());
+                        });
+
+                TDecorators.wrap(channel.getOverrideMap().keySet()).stream()
+                        .map(id -> channel.getGuild().getMemberById(id))
+                        .filter(Objects::nonNull)
+                        .filter(user -> !containedMembers.contains(user))
+                        .forEach(member -> {
+                            changedMembers.add(member);
+                            toRemove.add(member.getUser().getIdLong());
+                        });
+
+                toRemove.forEach((id) -> {
+                    channel.getOverrideMap().remove(id);
+                    return true;
```

Comment:
This is part of trove collections, when true is returned the next iteration step starts otherwise the forEach is interrupted.